### PR TITLE
Twitter channels based on the beta Activity API

### PR DESCRIPTION
**Pull request by rowanseymour**
Only visible to beta users for now

screen shot 2017-06-22 at 14 17 46

screen shot 2017-06-22 at 14 17 34

### CODE REVIEW

File: templates/channels/channel_claim.haml [hunk 1]
```diff
@@ -636,104 +636,121 @@
...
 
-      .claim-row.twitter-channel
-        .row
-          .span3
-            %a.btn.btn-primary.btn-primary-muted.btn-claim{href:"{% url 'channels.channel_claim_telegram' %}"}
-              .btn-container
-                .claim-glyph.icon-telegram
-                %p.claim-text
-                  -trans "Telegram"
+    .claim-row.telegram-channel
```
Comment (rowanseymour) [inline templates/channels/channel_claim.haml]: All of these lines below were wrongly nested inside the h3 causing them to not display correctly

File: temba/channels/models.py [hunk 2]
```diff
@@ -758,6 +756,33 @@ def add_twitter_channel(cls, org, user, screen_name, handle_id, oauth_token, oau
...
+
+        with org.lock_on(OrgLock.channels):
+            channel = Channel.create(org, user, None, Channel.TYPE_TWITTER, name="@%s" % screen_name, address=screen_name, config=config)
+
+            def register_webook():
+                callback_url = 'https://%s%s' % (settings.HOSTNAME, reverse('handlers.twitter_handler', args=[channel.uuid]))
+                webhook = twitter.register_webhook(callback_url)
+                twitter.subscribe_to_webhook(webhook['id'])
+
+            on_transaction_commit(register_webook)
```
Comment (nicpottier) [inline temba/channels/models.py]: Should we just make these non-atomic? We are only inserting a single row anyways.

And do we need the lock? What is that preventing? Seems a double click will still lead to one of these being clobbered no?

Comment (rowanseymour) [inline temba/channels/models.py]: Oh we don't sorry that's just cruft - some of our add_x_channel methods follow a pattern of updating an existing channel if it exists or creating a new one (hence the lock) - but I think we're moving toward immutable channels right? So the form throws a validation error if you try to create two channels from the same handle

Comment (rowanseymour) [inline temba/channels/models.py]: What's the reason to make these methods non-atomic? No biggie to call on_transaction_commit

**Comment by nicpottier** [comment]
Argument for non-atomic is just simplicity of not having a clojure and celery task when the atomicity really isn't buying us anything.

**Comment by rowanseymour** [comment]
Fair nuff tho soon all the `Channel.add_??_channel` methods aren't going to exist!

**Pull request merged**